playbooks/roles/common/tasks/repositories.yml

```diff
@@ -1,9 +1,21 @@
 ---
 - name: 'Setup Epel Repository'
   yum: name=http://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ release }}.noarch.rpm state=present
-- name: 'Setup Puppetlabs Repository'
-  yum: name=http://yum.puppetlabs.com/puppetlabs-release-el-{{ release }}.noarch.rpm state=present
+
+- name: 'Setup Puppet 3 Repository'
+  yum:
+    name: http://yum.puppetlabs.com/puppetlabs-release-el-{{ release }}.noarch.rpm
+    state: present
+  when: common_puppet_version is undefined or common_puppet_version == 3
+
+- name: 'Setup Puppet 4 Repository'
+  yum:
+    name: https://yum.puppetlabs.com/puppetlabs-release-pc1-el-{{ release }}.noarch.rpm
+    state: present
+  when: common_puppet_version is defined and common_puppet_version == 4
```

Comment:
I get the opposite of kludgy -- many small roles doing one dedicated thing makes it easy to then combine in a playbook. For me, thats what brings it all together. Small dedicated roles that you use playbooks to combine into a larger orchestration. The alternative is you create few larger playbooks with lots of variables and conditionals which to me would be harder to follow when looking at it. The reason I have to shy away from a common role is what happens if everybody has a common role and you start using third party roles? Clash of the titans.